File: app/src/main/java/cat/xojan/random1/ui/BaseFragment.java
package cat.xojan.random1.ui;

import android.support.v4.app.Fragment;

import cat.xojan.random1.injection.HasComponent;

public abstract class BaseFragment extends Fragment {

    /**
     * Gets a component for dependency injection by its type.
     */
    @SuppressWarnings("unchecked")
    protected <C> C getComponent(Class<C> componentType) {
        return componentType.cast(((HasComponent<C>) getActivity()).getComponent());
    }
}

File: app/src/main/java/cat/xojan/random1/ui/BaseFragment.kt
package cat.xojan.random1.ui

import android.support.v4.app.Fragment
import android.support.v4.media.MediaBrowserCompat

import cat.xojan.random1.injection.HasComponent

abstract class BaseFragment : Fragment() {

    /**
     * Gets a component for dependency injection by its type.
     */
    protected fun <C> getComponent(componentType: Class<C>): C =
            componentType.cast((activity as HasComponent<*>).component)

    /**
     * Override for custom support for back button click event.
     */
    fun handleOnBackPressed(): Boolean {
        return false
    }
}
